### ISSUE TITLE
Feature Request: Method to reset *just* optimizer

### ISSUE THREAD

**Issue by stevenwong** [keras-team/keras#11699]
Hi all,

I have a simple feature request - is it possible to just reset the state of the optimizer (i.e., momentum)?

I know I can do so with `model.compile`, but it's rather slow and I don't need to reset other stuff like loss function etc.

My use case is this: I'm designing a custom training scheme where I need to reset the state of the optimizer when certain conditions are met, not exactly the use case of a learning rate scheduler.

What do you guys think?

Cheers,
Steve

Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
